### PR TITLE
Fix re-export and docs for identity

### PR DESCRIPTION
**Pull request by jlperla**
# What does this PR do?
The addition of #4652  didn't properly re-export the function as `nnx.identity` and didn't end up in the documentation.

Dscussed in a Github issue #4635

### CODE REVIEW

**Comment by jlperla** [comment]
@IvyZX   Some context here:  this should be a non-breaking change, and requires no modification of the JAX version.  The previous PR in #4652  was hard for to test docs because the JAX change hadn't merged quite yet - but now it has.

File: docs_nnx/api_reference/flax.nnx/nn/activations.rst [hunk 1]
```diff
@@ -21,6 +21,7 @@ Activation functions
 .. autofunction:: relu6 as relu6,
 .. autofunction:: selu
 .. autofunction:: sigmoid
+.. autofunction:: identity
```
Comment (IvyZX) [inline docs_nnx/api_reference/flax.nnx/nn/activations.rst]: Please remove the trailing whitespace in this line.

**Review by IvyZX — CHANGES_REQUESTED** [review]
Looks good to me! Just remove the whitespace and make the CI test pass, then I'll merge it.

**Comment by jlperla** [comment]
Sorry this tiny change has been a pain @IvyZX   
I am not entirely sure, but looks like the doctest is whitespace sensitive and it is breaking? https://github.com/google/flax/actions/runs/16581993163/job/46900333462#step:7:982 

It says
```
File "../.venv/lib/python3.11/site-packages/jax/_src/nn/functions.py", line ?, in default
Failed example:
    jax.nn.identity(jax.numpy.array([-2., -1., -0.5, 0, 0.5, 1., 2.]))
Expected:
    Array([-2. , -1. , -0.5, 0. , 0.5, 1. , 2. ], dtype=float32)
Got:
    Array([-2. , -1. , -0.5,  0. ,  0.5,  1. ,  2. ], dtype=float32)
```

Which looks like the jax: https://github.com/jax-ml/jax/blob/5067a1860cdb33bee56257c9dc9f1f37198c2584/jax/_src/nn/functions.py#L67-L68 has introduced some sort of extra whitespace, but the jax repo itself doesn't run doctest so it wasn't seen before?

If so, then it looks like I need to change https://github.com/jax-ml/jax/blob/5067a1860cdb33bee56257c9dc9f1f37198c2584/jax/_src/nn/functions.py#L68 to include that extra space,  and then we can only merge this PR after a new JAX version?

Or am I missing something?

**Comment by IvyZX** [comment]
I noticed this error didn't happen in JAX documentation, which is likely because they had the `NORMALIZE_WHITESPACE` [pytest flag](https://github.com/jax-ml/jax/blob/1ab0f0ed1d85ec8cc11c44509b1deb14758c81df/pyproject.toml#L91).

We do want to align with JAX docs. Could you help add the following in Flax `pyproject.toml`'s `pytest` options [here](https://github.com/google/flax/blob/main/pyproject.toml#L154) and see if that fixes it? 
```
doctest_optionflags = [
    "NUMBER",
    "NORMALIZE_WHITESPACE"
]
```

**Comment by jlperla** [comment]
@IvyZX  Seems to have worked!

**Comment by jlperla** [comment]
Sorry @IvyZX  I think I spoke too soon as it looks like my addition to the `pyproject.toml` didn't fix the problem?

**Comment by jlperla** [comment]
@IvyZX  Sorry to bump this.  I am still not sure the next steps given your ask to

>  Could you help add the following in Flax pyproject.toml's pytest options [here](https://github.com/google/flax/blob/main/pyproject.toml#L154) and see if that fixes it?`

**Comment by IvyZX** [comment]
Apologies, I didn't have time to get back to this. The error seems enigmatic, and I don't know why it worked with JAX documentation but not Flax's. I can keep investigating.

**Comment by IvyZX** [comment]
@jlperla Hey - I sent you a PR that will fix this error (locally tested): https://github.com/jlperla/flax/pull/1

Feel free to merge it and retry here, and then I can approve your branch.

**Comment by jlperla** [comment]
Thanks @IvyZX 

I merged but wasn't sure how to trigger the CI again

**Review by IvyZX — APPROVED** [review]
